File: app/src/main/java/com/deepconverse/webviewapp/MainActivity.java
package com.deepconverse.webviewapp;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;

import androidx.appcompat.app.AppCompatActivity;

import com.deepconverse.android_sdk.DeepConverseSDK;
import com.google.gson.Gson;

import java.util.HashMap;
import java.util.Map;


public class MainActivity extends AppCompatActivity implements DeepConverseSDK.WebViewCallback {

    private Button openWebViewButton;
    private LinearLayout webUrlContainer;
    private DeepConverseSDK deepConverseSDK;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        openWebViewButton = findViewById(R.id.openWebViewButton);
        webUrlContainer = findViewById(R.id.webUrlContainer);

        openWebViewButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (deepConverseSDK != null) {
                    // Remove the existing WebUrlView if present
                    webUrlContainer.removeView(deepConverseSDK);
                    deepConverseSDK.destroyView();
                }

                // Create a new instance of WebUrlView
                Map<String, String> metadata = new HashMap<>();
                Gson gson = new Gson();//
                metadata = gson.fromJson("{\"country\":\"US\", \"email\": \"bob@ose.io\"}", Map.class);
                metadata.put("draft", "true");
                deepConverseSDK = new DeepConverseSDK(MainActivity.this, "dcstg5",
                        "preshin-19", metadata);
                deepConverseSDK.setLayoutParams(new LinearLayout.LayoutParams(
                        LinearLayout.LayoutParams.MATCH_PARENT,
                        LinearLayout.LayoutParams.MATCH_PARENT));
                deepConverseSDK.setWebViewCallback(MainActivity.this);
                deepConverseSDK.load();
                webUrlContainer.addView(deepConverseSDK);

                // Show the webUrlContainer and trigger a layout pass
                webUrlContainer.setVisibility(View.VISIBLE);
                webUrlContainer.requestLayout();
            }
        });
    }

    @Override
    public void onViewClosed() {
        // Remove the WebUrlView from the container
        if (deepConverseSDK != null) {
            webUrlContainer.removeView(deepConverseSDK);
            deepConverseSDK.destroyView();
            deepConverseSDK = null;
            webUrlContainer.setVisibility(View.GONE);
        }
    }
}

File: app/src/main/java/com/deepconverse/www/MainActivity.kt
package com.deepconverse.www

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import com.deepconverse.chatbot.DeepConverseCallbacks
import com.deepconverse.chatbot.DeepConverseErrors
import com.deepconverse.chatbot.DeepConverseSDK
import com.deepconverse.chatbot.DeepConverseSession

class MainActivity : AppCompatActivity(), DeepConverseCallbacks {

    private lateinit var deepConverseSDK: DeepConverseSDK
    private lateinit var button: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val items = HashMap<String, String>()
        items["draft"] = "true"

        val session = DeepConverseSession(
            subdomian = "dcshow1",
            botname = "showbot",
            metadata = items
        )
        deepConverseSDK = DeepConverseSDK(
            session = session,
            callbacks = this
        )
        button = findViewById(R.id.open_button)
        button.setOnClickListener {
            open()
        }
    }

    private fun open() {
        deepConverseSDK.open(context = this)
    }

    override fun didCloseBot() {
    }

    override fun didOpenBot() {
    }

    override fun didFailToLoadBot(errors: DeepConverseErrors) {
    }

    override fun didReceiveEvent(event: Map<String, String>) {
    }
}